src/controllers/book.ts

```diff
@@ -1,18 +1,16 @@
-import { inject, injectable } from 'tsyringe';
-import Book from '../database/schemas/Book';
-import { BookInterface } from '../interfaces/book';
-import IBookRepository from '../interfaces/repositories/IBook';
+import { Response, Request } from 'express';
+import { container } from 'tsyringe';
+
+import BookService from '../services/book';
 
-@injectable()
 export default class BookController {
-    constructor(
-        @inject('BookRepository')
-        private bookRepository: IBookRepository,
-    ) {}
+    public async create(request: Request, response: Response): Promise<Response> {
+        const book_data = request.body;
+
+        const bookService = container.resolve(BookService);
```

Comment:
Instantiate this property in the constructor so you can use the same instance of bookService in all functions of this controller.